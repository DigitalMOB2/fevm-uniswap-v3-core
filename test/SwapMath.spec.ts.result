

  SwapMath
    #computeSwapStep
      ✔ exact amount in that gets capped at price target in one for zero
      ✔ exact amount out that gets capped at price target in one for zero
      ✔ exact amount in that is fully spent in one for zero
      ✔ exact amount out that is fully received in one for zero
      ✔ amount out is capped at the desired amount out
      ✔ target price of 1 uses partial input amount
      ✔ entire input amount taken as fee
      ✔ handles intermediate insufficient liquidity in zero for one exact output case
      ✔ handles intermediate insufficient liquidity in one for zero exact output case
      gas
        ✔ swap one for zero exact in capped
        ✔ swap zero for one exact in capped
        ✔ swap one for zero exact out capped
        ✔ swap zero for one exact out capped
        ✔ swap one for zero exact in partial
        ✔ swap zero for one exact in partial
        ✔ swap one for zero exact out partial
        ✔ swap zero for one exact out partial


  17 passing (5s)

