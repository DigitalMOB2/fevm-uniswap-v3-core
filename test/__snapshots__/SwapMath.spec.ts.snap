// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwapMath #computeSwapStep gas swap one for zero exact in capped 1`] = `596496`;

exports[`SwapMath #computeSwapStep gas swap one for zero exact in partial 1`] = `725268`;

exports[`SwapMath #computeSwapStep gas swap one for zero exact out capped 1`] = `530516`;

exports[`SwapMath #computeSwapStep gas swap one for zero exact out partial 1`] = `725268`;

exports[`SwapMath #computeSwapStep gas swap zero for one exact in capped 1`] = `633408`;

exports[`SwapMath #computeSwapStep gas swap zero for one exact in partial 1`] = `976938`;

exports[`SwapMath #computeSwapStep gas swap zero for one exact out capped 1`] = `567428`;

exports[`SwapMath #computeSwapStep gas swap zero for one exact out partial 1`] = `976938`;
